### DISCUSSION TITLE
Go Benchmarks in Github Actions

### DISCUSSION THREAD

**Discussion by embano1** [timbray/quamina#139] [Ideas]
While the newly added benchmark GHA running on PRs to detect regressions gives us some good insights, it is currently hard to parse as we cannot directly alert on a PR [due to Github's security model](https://github.com/timbray/quamina/pull/126#issuecomment-1296323463) for workflows triggered on `pull_request`.

IMHO we could simplify the whole benchmark workflow and make use of [Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) **if** we can get a nicely rendered markdown table from the benchmark output. I am using [`tparse`](https://github.com/mfridman/tparse/actions/runs/2837125980) to do this for regular `go test` but haven't looked into it wrt `go test -bench`.

I am wondering if we could just adopt `tparse` or something custom we write to get a markdown table for benchmarks which we print on the job summary and have it fail on threshold X. Thoughts? @timbray @yosiat

**Comment by embano1** [comment]
Update: there's a tracking issue in `tparse`: https://github.com/mfridman/tparse/issues/67

**Comment by timbray** [comment]
Micha, I'm going to trust you on this one, I am still way behind on this CI stuff (catching up a bit, but still). I would like us to have all non-failing actions though.

**Reply by embano1** [reply]
Sure, will keep track of it so at least we simplify the benchmark review experience (using summaries).

**Comment by yosiat** [comment]
@embano1 Sounds good to me 👍
